antarest/study/business/binding_constraint_management.py

```diff
@@ -642,38 +650,36 @@ def create_binding_constraint(
 
         check_attributes_coherence(data, version)
 
-        new_constraint = {
-            "name": data.name,
-            "enabled": data.enabled,
-            "time_step": data.time_step,
-            "operator": data.operator,
-            "coeffs": self.terms_to_coeffs(data.terms),
-            "values": data.values,
-            "less_term_matrix": data.less_term_matrix,
-            "equal_term_matrix": data.equal_term_matrix,
-            "greater_term_matrix": data.greater_term_matrix,
-            "filter_year_by_year": data.filter_year_by_year,
-            "filter_synthesis": data.filter_synthesis,
-            "comments": data.comments or "",
-        }
+        new_constraint = {"name": data.name}
+        properties = list(data.dict(exclude={"terms", "name", "operator", "time_step"}).keys())
```

Comment:
It seems that this code is duplicated between creation and update

Reply:
Well, we could try to avoid that…

Reply:
I think we can simplify:

```python
new_constraint = {"name": data.name, **json.loads(data.json(exclude={"terms", "name"}, exclude_none=True))}
```

Reply:
Really nice :)